test/unit/test_connection.py

```diff
@@ -136,12 +136,46 @@ def test_is_still_running():
 
 
 @pytest.mark.skipolddriver
-def test_partner_env_var():
-    with patch.dict(os.environ, {ENV_VAR_PARTNER: "Amanda"}):
-        with patch("snowflake.connector.network.SnowflakeRestful.fetch"):
-            with snowflake.connector.connect(
-                user="user",
-                account="account",
-                password="password123",
-            ) as conn:
-                assert conn.application == "Amanda"
+@patch("snowflake.connector.network.SnowflakeRestful._post_request")
+def test_partner_env_var(mockSnowflakeRestfulPostRequest, capsys):
+    PARTNER_NAME = "Amanda"
+
+    def mock_post_request(url, headers, json_body, **kwargs):
+        global mock_cnt
+        print(json_body)
+        ret = None
+        if mock_cnt == 0:
+            # return from /v1/login-request
+            ret = {
```

Comment:
```suggestion
            return {
```
and then you could remove the dedented return statement below

Reply:
Good call. I copied this from a test above.

Reply:
I know some of the old tests are not great, I'm not picking on you, I swear I'll fix them all one day :)